kore/src/Kore/Internal/Variable.hs

```diff
@@ -0,0 +1,27 @@
+{-|
+Copyright   : (c) Runtime Verification, 2018
+License     : NCSA
+
+-}
+
+module Kore.Internal.Variable
+    ( InternalVariable
+    , module Kore.Syntax.ElementVariable
+    , module Kore.Syntax.SetVariable
+    , module Kore.Syntax.Variable
+    ) where
+
+import Kore.Debug
+       ( Debug )
+import Kore.Syntax.ElementVariable
+import Kore.Syntax.SetVariable
+import Kore.Syntax.Variable
+import Kore.Unparser
+       ( Unparse )
+
+-- | 'InternalVariable' is the constraint on internal variable types.
```

Comment:
I wouldn't know when to use this constraint by reading this documentation (and, since I didn't carefully read the entire PR, I actually don't know what it means).

So, I guess that "SimplifierVariable" should be used in the simplifier. Should we use "InternalVariable" in all other cases, i.e. the ones that don't need the FreshVariable constraint?

Feel free to ignore if it does not make sense.

Reply:
I added comments to each constraint type explaining when we would use each.